src/app/static/src/components/Collapsible.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div>
+    <h3 @click="toggleOpen" class="cursor-pointer">
+      {{heading}}
+      <component style="vertical-align: initial"
+                 :is="chevronComponent"></component>
+    </h3>
+    <b-collapse v-model="open">
+      <slot></slot>
+    </b-collapse>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { ChevronDownIcon, ChevronUpIcon } from "vue-feather-icons";
+import { BCollapse } from "bootstrap-vue";
+
+interface Methods {
+    toggleOpen: () => void
+}
+
+interface Props {
+    initialOpen: boolean,
+    heading: string
+}
+
+interface Data {
+    open: boolean
+}
+
+export default Vue.extend<Data, Methods, unknown, Props>({
```

Comment:
Genuine question: it is necessary/useful to specify `Props` here even if we're already specifying the types inline below?

Reply:
Typescript compilation fails when using `this.someProp` if the Props are not defined.